src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ConversationService.java

```diff
@@ -360,12 +379,35 @@ public Page<User> searchMembersOfConversation(Course course, Conversation conver
      *
      * @param conversationId the id of the conversation
      * @param requestingUser the user that wants to switch the favorite status
-     * @param isFavorite     the new favorite status
+     * @param favoriteStatus the new favorite status
      */
-    public void switchFavoriteStatus(Long conversationId, User requestingUser, Boolean isFavorite) {
-        var participation = conversationParticipantRepository.findConversationParticipantByConversationIdAndUserIdElseThrow(conversationId, requestingUser.getId());
-        participation.setIsFavorite(isFavorite);
-        conversationParticipantRepository.save(participation);
+    public void switchFavoriteStatus(Long conversationId, User requestingUser, Boolean favoriteStatus) {
+        var participation = conversationParticipantRepository.findConversationParticipantByConversationIdAndUserId(conversationId, requestingUser.getId());
+
+        if (participation.isEmpty()) {
+            Conversation conversation = conversationRepository.findByIdElseThrow(conversationId);
+
+            if (conversation instanceof Channel channel && channel.getIsCourseWide()) {
+                ConversationParticipant conversationParticipant = new ConversationParticipant();
+                conversationParticipant.setUser(requestingUser);
+                conversationParticipant.setConversation(conversation);
+                conversationParticipant.setIsModerator(false);
+                conversationParticipant.setIsHidden(false);
+                conversationParticipant.setIsFavorite(favoriteStatus);
+                conversationParticipant.setLastRead(ZonedDateTime.now().minusYears(10));
+                conversationParticipant.setUnreadMessagesCount(0L);
+                conversationParticipantRepository.save(conversationParticipant);
```

Comment:
This part seems duplicated, can you move this to a method / constructor of ConversationParticipant?

Reply:
I moved it into a method in ConversationParticipant